modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,193 @@
+    }
+    return this.serialize();
+  }
+
+  /** @inheritdoc */
+  toJson(): TxData {
+    if (!this._atomTransaction) {
+      throw new ParseTransactionError('Empty transaction');
+    }
+    const tx = this._atomTransaction;
+    return {
+      id: this._id,
+      type: tx.type,
+      signerAddress: tx.signerAddress,
+      explicitSignerData: tx.explicitSignerData,
+      sendMessage: tx.sendMessage,
+      gasBudget: tx.gasBudget,
+    };
+  }
+
+  /** @inheritDoc */
+  explainTransaction(): TransactionExplanation {
+    const result = this.toJson();
+    const displayOrder = ['id', 'outputs', 'outputAmount', 'changeOutputs', 'changeAmount', 'fee', 'type'];
+    const outputs: TransactionRecipient[] = [];
+
+    const explanationResult: TransactionExplanation = {
+      displayOrder,
+      id: this.id,
+      outputs,
+      outputAmount: '0',
+      changeOutputs: [],
+      changeAmount: '0',
+      fee: { fee: this.atomTransaction.gasBudget.gas.toString() }, // TODO - explore the fee explanation more, do we need to show limit or base amount
+      type: this.type,
+    };
+
+    switch (this.type) {
+      case TransactionType.Send:
+        return this.explainTransferTransaction(result, explanationResult);
+      default:
+        throw new InvalidTransactionError('Transaction type not supported');
+    }
+  }
+
+  /**
+   * Set the transaction type.
+   *
+   * @param {TransactionType} transactionType The transaction type to be set.
+   */
+  transactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: string): void {
+    try {
+      this._atomTransaction = Transaction.deserializeAtomTransaction(rawTransaction);
+      this._type = TransactionType.Send;
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  serialize(): string {
+    // TODO BG-67032 - if possible use Stargate Client builder to build [and sign, if applicable] tx using data types from this tx class
+    // Needs research: how to get stargate client builder without feeding an OfflineSigner private key
+    throw new Error('not yet implemented');
+  }
+
+  static deserializeAtomTransaction(txHex: string): AtomTransaction {
+    const decodedTx = decodeTxRaw(fromBase64(txHex));
+    const registry = new Registry(defaultRegistryTypes);
+    // TODO - support many messages decoding, right now we are just taking one message for send tx type
+    const encodedMessages = decodedTx.body.messages as EncodeObject[];
+    const decodedMessage = registry.decode(encodedMessages[0]); // TODO - do we need this?
+    let type: AtomTransactionType;
+    if (encodedMessages[0].typeUrl === sendMsgType) {
+      type = AtomTransactionType.Pay;
+    } else {
+      throw new Error('Transaction type not supported: ' + encodedMessages[0].typeUrl);
+    }
+    const sendMessageData: MessageData = {
```

Comment:
Could we have a helper method which does this `MessageData` construction for us instead of the `deseralizeAtomTransaction` method doing it ?

Reply:
Yes, this was just an initial draft, will be modularizing it today.